tests/api-compatibility/README.md

```diff
@@ -0,0 +1,126 @@
+        ApiCompatPath="../../packages/Microsoft.DotNet.ApiCompat.5.0.0-beta.19562.5/tools/net472/"
+        ApiLevel="$(AndroidFrameworkVersion)"
+        LastStableApiLevel="$(AndroidLatestStableFrameworkVersion)"
+        TargetImplementationPath="$(OutputPath)"
+        ApiCompatibilityPath = "$(ApiCompatibilityDir)" />
+    <Touch Files="$(IntermediateOutputPath)CheckApiCompatibility.stamp" AlwaysCreate="True" />
+    <ItemGroup>
+      <FileWrites Include="$(IntermediateOutputPath)CheckApiCompatibility.stamp" />
+    </ItemGroup>
+  </Target>
+```
+
+`ApiCompatibilityFiles` - Are the files we would like to monitor for changes, this is doen to prevent the task to run if it is not required. (there were no changes)
+
+`Touch Files` - It is our timestamp file to control if task should run.
+
+[bst]: https://github.com/xamarin/xamarin-android/tree/master/build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks
+
+## ApiCompatibilityPath
+
+On this path we will place the files that contain all acceptable breakages for a given Api version.
+
+for example:
+
+```
+acceptable-breakages-v7.0.txt
+acceptable-breakages-v7.1.txt
+acceptable-breakages-v8.0.txt
+acceptable-breakages-v9.0.txt
+```
+
+`acceptable-breakages-v7.0.txt` - Contains any rules that are accept to be broken in relation to the previous Api level of v7.0 (in this case v6.0)
+
+Api Level v6 used to have thefollow piece of code:
+
+```
+[Obsolete (@"deprecated")]
+public virtual unsafe Java.Lang.Object LastNonConfigurationInstance {
+```
+
+however Api level v7, removed the `[Obsolete (@"deprecated")]`
+```
+public virtual unsafe Java.Lang.Object LastNonConfigurationInstance {
+```
+
+The Api compat tool would report that as an Api breakage and cause the build to fail.
+
+However on file `acceptable-breakages-v7.0.txt` we added the folowing line:
+```
+CannotRemoveAttribute : Attribute 'System.ObsoleteAttribute' exists on 'Android.App.Activity.LastNonConfigurationInstance' in the contract but not the implementation.
+```
+that will cause the error to no cause a build failure.
+
+## Assembly reference
+
+Besides checking against a previous Api Level, we should also check against the current Api Level. This will prevent we break the current last stable api level. In order to be able to do that we need to commit the Mono.Android.dll assembly to our repository.
+
+We don't want to commit the "naked" assembly because it is quite large (around 30MB), so we should first:
+
+`Strip IL code out of the assembly` -> We do that using a tool called `cil-strip.exe
+On a mac machine run the follow command:
+
+```
+mono bin/Debug/lib/xamarin.android/xbuild/Xamarin/Android/cil-strip.exe Mono.Android.dll Mono.Android-out.dll
+Mono CIL Stripper
+
+26747392  Mono.Android.dll
+19300864  Mono.Android-out.dll
+```
+
+Rename Mono.Android-out.dll to Mono.Android.dll, compress (zip Mono.Android.dll.zip Mono.Android.dll)
+
+```
+4145708  Mono.Android.dll.zip
```

Comment:
We should *probably* provide an MSBuild target to do this, e.g.

```
msbuild src/Mono.Android/Mono.Android.csproj /t:UpdateReferenceAssembly
```

Additionally, this doc should mention *where* these strip'd-and-compressed files should *go*.

Based on `.gitignore`, presumably the files should go into `tests/api-compatibility/reference/*.dll.zip`?

Reply:
Where is `Mono.Android.dll.zip` *used from*?  I don't see anything within this PR to *unzip* the file so that it can be used by the tooling.

Reply:
The compressed files are unzipped during the check process. I believe you missed, but on CheckApiCompatibility.cs (line 93) I have this code:
```
// Before validate, check that zip files were decompressed.
var zipFiles = Directory.GetFiles(referenceContractPath, "*.zip");
foreach (var zipFile in zipFiles) {
	using (var zip = ZipArchive.Open (zipFile, FileMode.Open)) {
		zip.ExtractAll (referenceContractPath);
	}
}
```